workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/Injection/MonoBehaviourActivationManager.cs

```diff
@@ -104,26 +101,49 @@ public void EnableSpatialOSBehaviours()
 
             foreach (var behaviour in behavioursToEnable)
             {
-                behaviour.enabled = true;
+                RunWithExceptionHandling(() => behaviour.enabled = true);
+                enabledBehaviours.Add(behaviour);
             }
 
             behavioursToEnable.Clear();
         }
 
         public void DisableSpatialOSBehaviours()
         {
-            foreach (var behaviour in behavioursToDisable)
+            DisableAllSpatialOSBehavioursOfASet(behavioursToDisable);
+            behavioursToDisable.Clear();
+        }
+
+        private void DisableAllEnabledSpatialOSBehaviours()
+        {
+            DisableAllSpatialOSBehavioursOfASet(enabledBehaviours);
+        }
+
+        private void DisableAllSpatialOSBehavioursOfASet(HashSet<MonoBehaviour> behaviours)
```

Comment:
Nit: Can probably just rename to `DisableSpatialOSBehaviours`. The argument provides the context of the set

Reply:
I renamed the method to `DisableAllSpatialOSBehavioursInternal` so people don't confuse it over the other public `DisableSpatialOSBehaviours` method, intended to be called from outside.